lib/puppet/http/client.rb

```diff
@@ -150,6 +150,10 @@ def build_resolvers
       resolvers << Puppet::HTTP::Resolver::SRV.new(self, domain: Puppet[:srv_domain])
     end
 
+    if Puppet[:server_list] && !Puppet[:server_list].empty?
```

Comment:
This may need to be ` Puppet.settings.value(:server_list, Puppet[:environment].to_sym, true)`